### PR TITLE
Improve README and add some comments in values

### PR DESCRIPTION
**Pull request by dimitrigraf**
In the README.md
- Add a chapter on how to use extraConfig for S3 storage configuration
- Complete the list of existingSecret credential keys

In the values.yaml
- Add some comments on the use of existingSecret

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -365,14 +365,53 @@ Rather than specifying passwords and secrets as part of the Helm release values,
...
-| `email_password`       | SMTP user password                                     | Yes, but the value may be left blank if not required                                  |
-| `napalm_password`      | NAPALM user password                                   | Yes, but the value may be left blank if not required                                  |
-| `redis_tasks_password` | Password for the external Redis tasks database         | If `redis.enabled` is `false` and `tasksRedis.existingSecretName` is unset            |
-| `redis_cache_password` | Password for the external Redis cache database         | If `redis.enabled` is `false` and `cachingRedis.existingSecretName` is unset          |
-| `secret_key`           | Django session and password reset token encryption key | Yes, and should be 50+ random characters                                              |
+| Key                    | Description                                                   | Required?                                                                                         |
```
Comment (dimitrigraf) [inline README.md:368]: I didn't change the existing entries, just added some new ones:

- ldap_bind_password
- superuser_password
- superuser_api_token

File: README.md [hunk 2]
```diff
@@ -365,14 +365,53 @@ Rather than specifying passwords and secrets as part of the Helm release values,
...
+| `email_password`       | SMTP user password                                            | Yes, but the value may be left blank if not required                                              |
+| `ldap_bind_password`   | Password for LDAP bind DN                                     | If `remoteAuth.enabled` is `true` and `remoteAuth.backend` is `netbox.authentication.LDAPBackend` |
+| `napalm_password`      | NAPALM user password                                          | Yes, but the value may be left blank if not required                                              |
+| `redis_tasks_password` | Password for the external Redis tasks database                | If `redis.enabled` is `false` and `tasksRedis.existingSecretName` is unset                        |
+| `redis_cache_password` | Password for the external Redis cache database                | If `redis.enabled` is `false` and `cachingRedis.existingSecretName` is unset                      |
+| `secret_key`           | Django secret key used for sessions and password reset tokens | Yes                                                                                               |
```
Comment (dimitrigraf) [inline README.md:376]: Is it correct that secret_key, superuser_password and superuser_api_token are required? I didn't test thoroughly but guess so from the helm templating logic.

File: README.md [hunk 3]
```diff
@@ -365,14 +365,53 @@ Rather than specifying passwords and secrets as part of the Helm release values,
...
+And the secret then has to be referenced like this:
+
+```yaml
+extraConfig:
+  - secret: # same as pod.spec.volumes.secret
+      secretName: netbox-extra
+      items:
+        - key: s3-config.yaml
+          path: s3-config.yaml
+      optional: false
```
Comment (bootc) [inline README.md:413]: These 4 lines are actually superfluous and could be removed. You only need `items` if you want to expose a subset of the entries from the Secret, or want them to appear with a different name, and `optional` is false by default. Probably clearer to remove them.

File: README.md (hunk 2 above)
Comment (bootc) [reply README.md:376]: Yes, they are currently required. `secret_key` certainly is necessary all the time; the `superuser_*` entries are only actually used the first time NetBox starts (when a superuser account doesn't exist) but are still currently required all the time.

File: README.md (hunk 1 above)
Comment (bootc) [reply README.md:368]: LGTM

File: README.md (hunk 2 above)
Comment (dimitrigraf) [reply README.md:376]: I'll resolve this comment, then :+1:

File: README.md (hunk 3 above)
Comment (dimitrigraf) [reply README.md]: done, see latest commit

**Comment by dimitrigraf** [comment]
Is it good like this now or do I still need to fix something? Also, do you have a rough estimate, when you are going to merge the development branch into the master branch and therefore release the new chart version officially?

**Comment by bootc** [comment]
I've just been a bit short on time really, but hoping to make a release soon.

**Comment by dimitrigraf** [comment]
@bootc let me know if I can support you in any way :+1:

**Pull request merged**